airflow/jobs.py

```diff
@@ -344,6 +344,11 @@ def _launch_process(result_queue,
         def helper():
             # This helper runs in the newly created process
             log = logging.getLogger("airflow.processor")
+
+            # redirect stdout/stderr to log
+            sys.stdout = StreamLogWriter(log, logging.INFO)
```

Comment:
Shouldn't StreamLogWriter implement `isatty` and `encoding` too?
This might be different between Python 2 and 3.
I know some python modules, for example, `colorama`, `pysparkling` are using these attributes.

Reply:
I don't think so, but if you can show me where that is actually done in those projects I can have look.

Reply:
`isatty` should be defined as `lambda : False` on redirected streams.
See https://stackoverflow.com/questions/1077113/how-do-i-detect-whether-sys-stdout-is-attached-to-terminal-or-not 
Some modules rely on this method.
See for example recent discussion on https://github.com/h2oai/sparkling-water/issues/324#issuecomment-341023550

Reply:
sure I can add it, although here I don't see the need directly

Reply:
thank you. yep there is no need in Airflow itself.
although code that Airflow runs may rely on this attribute.